### ISSUE TITLE
Possible option to give a reply to the student asking for help?

### ISSUE THREAD

**Issue by Shadow53** [Shadow53/student-queue#9]
This could pop up a chat box of sorts and allow the teacher to send a reply message to the student. Need to get more important things in place, but could be a cool addition.

**Comment by Shadow53** [comment]
Could also be an alert box or an HTML5 notification - probably the latter. Then on `msgReceive` or whatever the event will be called, it notifies the student with the response.

Bonus feature: send message to multiple students at once.

**Comment by Shadow53** [comment]
So here's the thing with this: clicking on the help request currently deletes the request. If I were to add the option to send a reply to a request, it would probably turn out so the the request could be clicked on and there would be two buttons: one to reply, one to delete. I'm not sure I want to implement this until there's some indication that it will be used enough to warrant the extra click.

**Comment by Shadow53** [comment]
Decided to look at this again. I could implement a "reply" button as either inside or next to the help request: clicking on the help request itself removes it, but clicking reply pops up a window that allows the teacher to reply to the message. This could be a socket.io "room" inside the queue "namespace", particular to the issue or student - if it were the issue, more students with similar issues could be added.

As part of this, a log of messages could be saved in a database table with columns for the student, the issue, the queue, the message itself, and the timestamp. Then if the window were closed and reopened or someone joined the chat, they could get the history.

This would require a bit of database refactoring to make it so that the issue could be tracked by ID number rather than the student ID, and I should probably implement issue #13 (archiving help requests) as part of this database refactoring as well.

As for keeping track of students, perhaps this would involve allowing students to log in so that they can be verified as them when they ask for help. This would mean adding more to the Teacher Admin page too....

TL:DR; This will be a lot of work.

**Comment by Shadow53** [comment]
That moment when you're so tired, you think "Close and comment" means "comment and close the window".
